### ISSUE TITLE
Please open source the teacher logits

### ISSUE THREAD

**Issue by sanyalsunny111** [microsoft/Cream#127]
Dear Authors,

Very impressive work. For reproducibility purposes could you please share the teacher logits files for all the teachers shown in this paper?

**Comment by wkcn** [comment]
Thanks for your attention to our work!

Sorry for the late reply.

We have uploaded the teacher logits files, which were generated by CLIP-ViT-Large/14-22k on ImageNet-22k [(Ref. the tutorial)](https://github.com/microsoft/Cream/blob/main/TinyViT/docs/SAVE_TEACHER_LOGITS.md).

You can download them by [azcopy](https://learn.microsoft.com/en-us/azure/storage/common/storage-use-azcopy-v10).
```bash
azcopy cp "https://tracking2019.blob.core.windows.net/tinyvit-teacher-logits/clip_224_22k_top100_logits_bs128/?sv=2020-04-08&st=2022-10-07T06%3A14%3A52Z&se=2022-10-21T06%3A14%3A00Z&sr=c&sp=rl&sig=RPycNtS5qTk77Z2nmyQeUHObEzVHrRCxtc41%2BalTN9A%3D" ./ --recursive
```

There are 1440 files (`90 epochs * 8 ranks * (key + value files)`) about 499 GB (18GB for keys, 481GB for values).

You can download the logits in the specific epoch.
For example, download the logits in the epoch 0:
```bash
azcopy cp "https://tracking2019.blob.core.windows.net/tinyvit-teacher-logits/clip_224_22k_top100_logits_bs128/logits_top100_epoch0/?sv=2020-04-08&st=2022-10-07T06%3A14%3A52Z&se=2022-10-21T06%3A14%3A00Z&sr=c&sp=rl&sig=RPycNtS5qTk77Z2nmyQeUHObEzVHrRCxtc41%2BalTN9A%3D" ./ --recursive
```

The expiration time is Oct. 21, 2022. We can extend the date if necessary : ).

**Comment by wkcn** [comment]
Close it. Feel free to re-open it if necessary for extending the date : )

**Comment by HaoWuSR** [comment]
Could you please extend the date？ Thanks a lot !

**Comment by wkcn** [comment]
> Could you please extend the date？ Thanks a lot !

Hi @HaoWuSR , the date has been extended.

**Comment by sanyalsunny111** [comment]
Hello Sir could you also share the logits of teacher based on datasets such as imagenet-1K and any other dataset?

**Comment by wkcn** [comment]
Hi @sanyalsunny111 , sorry that I did not prepare the teacher logits on ImageNet-1k for this codebase.

You can follow the [tutorial](https://github.com/microsoft/Cream/blob/main/TinyViT/docs/SAVE_TEACHER_LOGITS.md) to save the sparse logits.

**Comment by sanyalsunny111** [comment]
Thank you for responding @wkcn Could you please provide a running script that could be used to save sparse logits of CLIP-ViT-Large/14-22k with imagenet-1k? I am asking cause I see save_logits.py can use both imagenet-22k and imagenet-1k. Any feedback on this?

**Comment by wkcn** [comment]
Hi @sanyalsunny111, the following command is used to save sparse logits of CLIP-ViT-Large/14-22k on ImageNet-1k.
```bash
python -m torch.distributed.launch --nproc_per_node 8 save_logits.py --cfg configs/teacher/clip_vit_large_patch14_22k.yaml --data-path ./ImageNet --batch-size 128 --eval --resume checkpoints/clip_vit_large_patch14_22k.pth --opts DISTILL.TEACHER_LOGITS_PATH ./teacher_logits_1k AUG.MIXUP 0.8 AUG.CUTMIX 1.0 DATA.DATASET imagenet TRAIN.EPOCHS 300 DISTILL.LOGITS_TOPK 10
```

It needs the extra config options `AUG.MIXUP 0.8 AUG.CUTMIX 1.0 DATA.DATASET imagenet TRAIN.EPOCHS 300 DISTILL.LOGITS_TOPK 10` to save the 1k logits.

However, I recommend Swin-base/large 22kto1k or BEiT-base/large 22kto1k as the 1k teacher model, since CLIP-ViT-Large/14-22k is not finetuned on ImageNet-1k.

**Comment by sanyalsunny111** [comment]
Hi @wkcn Thank you very much for earlier responses. I have saved the teacher logits on teacher CLIP-ViT-Large/14-22k using ImageNet-1k. Could you please provide a pre-training script (with distillation) with saved teacher logits on ImageNet.

**Comment by wkcn** [comment]
Hi @sanyalsunny111, you can try this script:
```bash
python -m torch.distributed.launch --nproc_per_node 8 main.py --cfg configs/1k/tiny_vit_21m.yaml --data-path ./ImageNet --batch-size 128  --opts DISTILL.TEACHER_LOGITS_PATH ./teacher_logits_1k AUG.MIXUP 0.8 AUG.CUTMIX 1.0 DATA.DATASET imagenet TRAIN.EPOCHS 300 DISTILL.LOGITS_TOPK 10
```

**Comment by sanyalsunny111** [comment]
> 

Just a quick question where the best checkpoints and training logs will be saved?

**Comment by wkcn** [comment]
Hi @sanyalsunny111 , 
The checkpoint and log will be saved in the folder `./output` by default. You can change it by adding the command `  --output xxxx` when running the script.

**Comment by sanyalsunny111** [comment]
Hey @wkcn could you please share your code for the Linear probe with various datasets as shown in Table 5 of the paper?

**Comment by wkcn** [comment]
Hi @sanyalsunny111 , please refer to [torchvision](https://pytorch.org/vision/stable/datasets.html) and [Cross-Domain Few-Shot Learning (CD-FSL) Benchmark](https://github.com/IBM/cdfsl-benchmark).

**Comment by sanyalsunny111** [comment]
Hi @wkcn, I have used Linear Probe and a few shot codes from moco-v3 and CD-FSL, but it's not matching your results on Table 5. Is it possible for you to share your codes for the Linear probing and Few shot?

**Comment by BrendanCReidy** [comment]
> > Could you please extend the date？ Thanks a lot !
> 
> Hi @HaoWuSR , the date has been extended.

Hi @wkcn can you please extend the date?

**Comment by wkcn** [comment]
Hi @BrendanCReidy, thanks for your attention to our work!
The new link: 
```
azcopy cp "https://tracking2019.blob.core.windows.net/tinyvit-teacher-logits?sv=2020-04-08&st=2023-01-26T03%3A00%3A21Z&se=2023-02-09T03%3A00%3A00Z&sr=c&sp=rl&sig=lVTekKcg9SAvzF6AmyGJTIv%2FXNx42XLLqSYSfpxmWLY%3D" ./ --recursive
```

**Comment by Andrewymd** [comment]
Hi @wkcn can you please extend the date?

**Comment by wkcn** [comment]
HI @Andrewymd , thanks for your attention to our work!
Here is the new link. The storage size of the teacher logits (clip_22k) is 499 GB.

Please install azcopy (https://learn.microsoft.com/en-us/azure/storage/common/storage-use-azcopy-v10) before downloading the teacher logits.

```
azcopy copy "https://hopengscus7387599778.blob.core.windows.net/tinyvit-teacher-logits/clip_224_22k_top100_logits_bs128/?sv=2021-08-06&se=2023-09-14T05%3A49%3A17Z&sr=c&sp=rl&sig=8HyBKYcLikVGtunLnTXgmtn8Ej0JiQsZt%2BCbexC6kfQ%3D" "./" --overwrite=prompt --check-md5 FailIfDifferent --from-to=BlobLocal --recursive --log-level=INFO
```

**Comment by jotoy** [comment]
Hi @wkcn can you please extend the date? Thanks for your great work~

**Comment by wkcn** [comment]
Hi @jotoy , here is the new download command.
```
azcopy copy  "https://hopengscus7387599778.blob.core.windows.net/tinyvit-teacher-logits/clip_224_22k_top100_logits_bs128/?sv=2021-08-06&st=2023-10-30T16%3A30%3A11Z&se=2023-11-29T16%3A30%3A00Z&sr=c&sp=rl&sig=MuAQWkYvVfD8q4YEKH2BdjdFoHR5ZkpKlMIOJlt2diM%3D" "./" --recursive
```

Thanks for your attention to our work!

**Comment by maybeliuchuan** [comment]
Hi @wkcn can you please extend the date? Thank you very much.

**Comment by maybeliuchuan** [comment]
Hi @wkcn, another question, since all of the 90epochs logits are too huge, whether can I train the TinyViT using only the logics of epoch 90? Thanks!

**Comment by wkcn** [comment]
@maybeliuchuan 
Thanks for your attention to our work!

Sorry that I could not extend the date. The logits can be reproduced by the teacher model CLIP-ViT-Large/14-22k on ImageNet-21k.

An epoch may be not enough on ImageNet-21k. You can try to save the teacher logits on ImageNet-1k. Here is the related tutorial: https://github.com/microsoft/Cream/blob/main/TinyViT/docs/SAVE_TEACHER_LOGITS.md
